pyface/util/color_parser.py

```diff
@@ -0,0 +1,340 @@
+        any additional names found in pyface.color.color_table.  The names
+        are normalized to lower case and stripped of whitespace, hyphens and
+        underscores.
+
+    Returns
+    -------
+    result : (space, channels), or None
+        Either a tuple of the form ('rgba', channels), where channels is a
+        tuple of 4 floating point values between 0.0 and 1.0, includive;
+        or None if there is no matching color name.
+    """
+    text = text.lower()
+    text = text.translate(ignored)
+    if text in color_table:
+        return 'rgba', color_table[text]
+    return None
+
+
+def parse_hex(text):
+    """ Parse a hex form of a color.
+
+    Parameters
+    ----------
+    text : str
+        A string holding a hex representation of the color in the form
+        '#RGB', '#RGBA', '#RRGGBB', '#RRGGBBAA', '#RRRRGGGGBBBB', or
+        '#RRRRGGGGBBBBAAAA'.
+
+    Returns
+    -------
+    result : (space, channels), or None
+        Either a tuple of the form (space, channels), where space is one of
+        'rgb' or 'rgba' and channels is a tuple of 3 or 4 floating point
+        values between 0.0 and 1.0, inclusive; or None if no hex
+        representation could be matched.
+    """
+    text = text.strip()
+    if re.match("#[0-9a-fA-F]+", text) is None:
+        return None
+    text = text[1:]
+    if len(text) in {3, 4}:
+        step = 1
+    elif len(text) in {6, 8}:
+        step = 2
+    elif len(text) in {12, 16}:
+        step = 4
+    else:
+        return None
+    maximum = (1 << 4 * step) - 1
+    channels = ints_to_channels(
+        (int(text[i:i+step], 16) for i in range(0, len(text), step)),
+        maximum=maximum,
+    )
+    space = 'rgb' if len(channels) == 3 else 'rgba'
+    return space, channels
+
+
+class ColorParseError(ValueError):
+    """ An Exception raised when parsing fails. """
+    pass
+
+
+def parse_text(text):
+    """ Parse a text representation of a color.
+
+    Parameters
+    ----------
+    text : str
+        A string holding the representation of the color.  This can be:
+
+        - a color name, including all CSS color names, plus any additional
+          names found in pyface.color.color_table.  The names are normalized
+          to lower case and stripped of whitespace, hyphens and underscores.
+
+        - a hex representation of the color in the form '#RGB', '#RGBA',
+          '#RRGGBB', '#RRGGBBAA', '#RRRRGGGGBBBB', or '#RRRRGGGGBBBBAAAA'.
+
+    Returns
+    -------
+    space : str
+        A string describing the color space for the channels.  Will be one of
+        'rgb' or 'rgba'.
+    channels : tuple of floats
+        The channel values as a tuple of 3 or 4 floating point values between
+        0.0 and 1.0, includive.
+
+    Raises
+    ------
+    ColorParseError
+        If the string cannot be converted to a valid color.
+    """
+    result = None
+    for parser in parse_name, parse_hex:
```

Comment:
That reminds me the "Speculative Generality" mentioned in the Code Smell chapter of the Refactoring book.
As a summary: Adding all sorts of hooks to handle things that aren't required results in code that is harder to understand and maintain.  If the machinery was needed, it would be worth it, but if it isn't, it isn't. The machinery gets in the way, so get rid of it.

The suggestion will only result in a limitation that none of the named color should start with `"#"`.  It is probably a good idea to never allow a name starts with `"#"` in order to parse hex string confidently.

Currently this is already costing user friendliness due to a less helpful error message.

> Probably should reverse the order, however.

If the order matters, that's a smell.